### ISSUE TITLE
Words for the sections

### ISSUE THREAD

**Issue by zignig** [cqparts/cqparts_webapi#3]
@fragmuffin  do you have any text you would like to put in /templates/landing  ? These get inserted as html fragments in to the landing page.

**Comment by fragmuffin** [comment]
@zignig I don't have any planned, but I'm sure I can think of something.

There's already plenty in the existing readmes, I'll put something together ;)

**Comment by fragmuffin** [comment]
@zignig Oh, no worries, I'll sort it out, I'll just translate my changes over to your templates, and static files.
I've just been having trouble running `serve.py`.

Do you mind if I set up a docker image to do the building / serving?
I tend to do everything in docker these days, because I've almost always regretted setting up my host in a particular way, because then there are some setup assumptions that get missed when deploying.

**Comment by zignig** [comment]
if you can manage it , 

I need to make a requirements  file at some point 

sqlalchemy , anytree , flask , requests 

and other sundry packages.

Luckily making the docker will _force_ you to get them right.

I currently use postgres as the backend , but a recent patch made it so you can use sqlite as your connection string.

Some pointers

./dump.py generates the website 

./blender/run.sh - runs the renderer ,  you will need https://github.com/ksons/gltf-blender-importer installed inside blender.

you  will need 

client_api.py:server = os.environ["CQPARTS_PROXY"] = http://localhost:8080
sql_string = os.environ["CQPARTS_DB"] = db connection, can be sqlite
app.secret_key = os.environ["CQPARTS_SECURE"] = random characters , used for cookies

set as environmental variables.

What issues are you having with server.py ?

**Comment by fragmuffin** [comment]
It'd be good to get TravisCI to do the deployment, so I think it's necessary.

> I currently use postgres as the backend , but a recent patch made it so you can use sqlite as your connection string.

Hmm, using postgres would make setup a bit harder, but is a better db overall...
wait what's the database for? (I haven't looked at how this works in detail yet)
Just for simplicity, I'll probably just go for `sqlite` in a docker container.

> What issues are you having with server.py ?

Oh! Just that things aren't installed.

I spent years writing tools for other engineers to use, but they often weren't software engineers, but systems or electrical, so expecting them to deal with setup issues wasn't realistic.
Instead I got into the habit of _only_ installing libraries on my host that were necessary for me to develop, and running the tools was done in a virtual environment (wherever practical).
Now with docker, conda, and python virtual environments, that's become easier... but the later 2 (conda & python envs), the fallback is always the host system.

TL;DR: To avoid any false-passes, I don't have `cadquery` or `cqparts`, or many of the dependent libraries installed on my host.

**Comment by zignig** [comment]
I'm out of town for a week on an install. If I get time , or when I return I will write a requirements doc and a description of the internals of the server program. 

Creating a docker would be great as it would lower the bar for people wanting to try out cqparts significantly.

**Comment by fragmuffin** [comment]
I already use docker containers to do `cqparts`' testing... one for unit & integration testing, and another for deployment testing.
It wouldn't be difficult for me to create one for `cqparts` and add it to [docker-hub](https://hub.docker.com/).

I think I'll wait for the working OCC version first though, there's a lot changing, including the overall interface in cqparts/cqparts#110

But that's a good future-proofing point... because there will be a docker-hub image soon, setting `cqparts_webapi` up in the meantime could use a docker image [`FROM cqparts-unittest:ubuntu-py3`](https://github.com/cqparts/cqparts/tree/master/tests/env/ubuntu-py3), and install `cqparts_webapi` specific packages to that.
